File: pset2/caesar/caesar.c
#include <string.h>
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, char** argv)
{
    if(argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }

    int k = atoi(argv[1]);

    string text = get_string("plaintext: ");
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if(isupper(text[i]))
        {
            text[i] = (((text[i] - 'A' + k) % 26) + 'A');
        }

        if(islower(text[i]))
        {
            text[i] = (((text[i] - 'a' + k) % 26) + 'a');
        }
    }

    printf("ciphertext: %s\n", text);
}

File: pset6/caesar/caesar.py
import sys
from cs50 import get_string

key = int(sys.argv[1])
text = get_string("plaintext: ")

for i in range(len(text)):
    if(text[i].isalpha()):
        if(text[i].islower()):
            text = text[:i] + chr(((ord(text[i]) - ord('a') + key) % 26) + ord('a')) + text[i + 1:]
        else:
            text = text[:i] + chr(((ord(text[i]) - ord('A') + key) % 26) + ord('A')) + text[i + 1:]

print("ciphertext: {0}".format(text))
